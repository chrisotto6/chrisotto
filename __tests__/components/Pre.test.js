import { render, fireEvent } from '@testing-library/react'
import Pre from '@/components/Pre'

it('renders Pre unchanged', () => {
  const { container } = render(<Pre />)
  expect(container).toMatchSnapshot()
})

it('button mouse events correctly', () => {
  const { getByTestId } = render(<Pre />)
  fireEvent.mouseEnter(getByTestId('pre'))
  fireEvent.mouseLeave(getByTestId('pre'))
})

it('button click events correctly', () => {
  jest.useFakeTimers()
  jest.spyOn(global, 'setTimeout')

  const mockClipboard = {
    writeText: jest.fn(),
  }

  global.navigator.clipboard = mockClipboard
  const { getByTestId } = render(<Pre />)
  fireEvent.mouseEnter(getByTestId('pre'))
  fireEvent.click(getByTestId('copy-button'))
  expect(mockClipboard.writeText).toHaveBeenCalledTimes(1)
  expect(setTimeout).toHaveBeenCalledTimes(1)
  expect(setTimeout).toHaveBeenLastCalledWith(expect.any(Function), 2000)
})
