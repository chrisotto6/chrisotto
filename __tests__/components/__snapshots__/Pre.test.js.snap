// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Pre unchanged 1`] = `
<div>
  <div
    class="relative"
    data-testid="pre"
  >
    <pre />
  </div>
</div>
`;
